homeassistant/components/mqtt/__init__.py

```diff
@@ -194,7 +193,7 @@ async def async_check_config_schema(
     hass: HomeAssistant, config_yaml: ConfigType
 ) -> None:
     """Validate manually configured MQTT items."""
-    mqtt_data = get_mqtt_data(hass)
+    mqtt_data = hass.data[DATA_MQTT]
```

Comment:
Can we add a type hint for the dataclass

Reply:
The type hint is part of HassKey https://github.com/home-assistant/core/pull/117899/files#diff-a75f3d209c64aeda975a4c9ceb8bdb4e6e0040ba9d7f0bb242e524879f314a9bR425


https://developers.home-assistant.io/blog/2024/05/01/improved-hass-data-typing?_highlight=hasskey

Reply:
Ah okay, should have looked a little further

Reply:
No worries 👍  Its new